internal/controller/providers/pg/applier.go

```diff
@@ -1040,9 +1043,89 @@ func (p *applier) reconcilePGBackupsSpec() (pgv2.Backups, error) {
 
 	newBackups.PGBackRest.Repos = pgBackrestRepos
 	newBackups.PGBackRest.Global = pgBackrestGlobal
+	logger := log.FromContext(ctx)
+	logger.Info("!!!!!!! ", "newbackups", newBackups)
 	return newBackups, nil
 }
 
+func getDataSourceStorage(ctx context.Context, c client.Client, db *everestv1alpha1.DatabaseCluster) (*everestv1alpha1.BackupStorage, *corev1.Secret, error) {
+	if db.Spec.DataSource == nil {
+		return nil, nil, errDataSourceNotFound
+	}
+	var dataSourceStorageName string
+	if db.Spec.DataSource.DBClusterBackupName != "" {
+		backup := &everestv1alpha1.DatabaseClusterBackup{}
+		err := c.Get(ctx, types.NamespacedName{Name: db.Spec.DataSource.DBClusterBackupName, Namespace: db.Namespace}, backup)
+		if err != nil {
+			return nil, nil, err
+		}
+		dataSourceStorageName = backup.Spec.BackupStorageName
+	} else {
+		dataSourceStorageName = db.Spec.DataSource.BackupSource.BackupStorageName
+	}
+
+	if dataSourceStorageName == "" {
+		return nil, nil, errBackupSourceNotFoundInDataSource
+	}
+
+	backupStorage := &everestv1alpha1.BackupStorage{}
+	err := c.Get(ctx, types.NamespacedName{Name: dataSourceStorageName, Namespace: db.Namespace}, backupStorage)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	backupStorageSecret := &corev1.Secret{}
+	key := types.NamespacedName{Name: backupStorage.Spec.CredentialsSecretName, Namespace: backupStorage.GetNamespace()}
+	err = c.Get(ctx, key, backupStorageSecret)
+	if err != nil {
+		return nil, nil, err
+	}
+	return backupStorage, backupStorageSecret, nil
```

Comment:
do we check storage's secret? this is not backup responsibility to sure that secret exists

Reply:
the purpose of getting the secret here is not to check if the secret exists (if for some reason it doesn't, everest just thows an error) but to configure the pgBackRest section for the pg cluster. Everest creates the secret named `database.Name+"-pgbackrest-secrets"` which contains the creds for each repo, that's why we need to pull the secret.